curated_transformers/models/pytorch/attention.py

```diff
@@ -8,28 +8,37 @@
 from ...errors import Errors
 
 
-@dataclass
 class AttentionMask:
     bool_mask: Tensor
-    _logit_mask: Optional[Tensor] = None
+    _logit_mask: Optional[Tensor]
 
-    def __post_init__(self):
-        if self.bool_mask.dtype != torch.bool:
-            raise ValueError(Errors.E005.format(dtype=self.bool_mask.dtype))
+    def __init__(self, bool_mask: Tensor):
+        if bool_mask.dtype != torch.bool:
+            raise ValueError("Expected the attention mask to be of dtype 'torch.bool'")
+        self.bool_mask = bool_mask
+
+        # Clarify to TorchScript JIT that we are not narrowing the type
+        # to `None`.
+        logit_mask: Optional[Tensor] = None
+        self._logit_mask = logit_mask
 
     @property
     def logit_mask(self) -> Tensor:
         if self._logit_mask is None:
             # The value is `torch.finfo(attn_scores.dype).min`. Unfortunately,
             # we cannot use `torch.finfo` in TorchScript.
             self._logit_mask = (1.0 - self.bool_mask.int()) * -3.4028234663852886e38
-        return self._logit_mask
+
+        # Narrow type for TorchScript.
+        logit_mask = self._logit_mask
+        assert logit_mask is not None
+        return logit_mask
 
     def dim(self) -> int:
         return self.bool_mask.dim()
 
     @property
-    def shape(self) -> Tuple:
+    def shape(self):
```

Comment:
A comment clarifying the missing type hint would be good to have here (since it's the odd one out).

Reply:
Same as the exceptions. Maybe a `DEVELOP.md` with a link to types that are permitted in TorchScript?